controllers/flashsystemcluster_controller.go

```diff
@@ -53,6 +58,49 @@ type SecretMapper struct {
 	reconciler *FlashSystemClusterReconciler
 }
 
+type CSIBlockMapper struct {
+	reconciler *FlashSystemClusterReconciler
+}
+
+var RunDeletePredicate = predicate.Funcs{
+	CreateFunc: func(e event.CreateEvent) bool {
+		return false
+	},
+	DeleteFunc: func(e event.DeleteEvent) bool {
+		return true
+	},
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		return false
+	},
+	GenericFunc: func(e event.GenericEvent) bool {
+		return false
+	},
+}
+
+func (s *CSIBlockMapper) CSIToClusterMapFunc(_ client.Object) []reconcile.Request {
+	clusters := &odfv1alpha1.FlashSystemClusterList{}
+
+	err := s.reconciler.Client.List(context.TODO(), clusters)
+	if err != nil {
+		s.reconciler.Log.Error(err, "failed to list FlashSystemCluster", "CSIToClusterMapFunc", s)
+		return nil
+	}
+
+	requests := []reconcile.Request{}
+	for _, c := range clusters.Items {
+		req := reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Namespace: c.GetNamespace(),
+				Name:      c.GetName(),
+			},
+		}
+		requests = append(requests, req)
+		s.reconciler.Log.Info("Discovered a change in CSI CR. Reconciling FlashSystemCluster", "CSIToClusterMapFunc", requests)
```

Comment:
consider moving this log outside the loop

Reply:
fixed